website/server/models/user/methods.js

```diff
@@ -42,6 +42,15 @@ schema.methods.addNotification = function addUserNotification (type, data = {})
   });
 };
 
+schema.methods.addNotificationUpdate = async function addUserNotificationUpdate (type, data = {}) {
+  let newNotification = {
+    type,
+    data,
+  };
+  this.notifications.push(newNotification);
```

Comment:
What we want to do with this method is to provide a way to add notifications without having to save the user. So this should probably be a static method of the user model (User.statics.addUserNotificationUpdate) that takes the notification data and one or more ids of users. If you issue the update command and add the data to the user doc (this.notifications.push) the next time the user is saved, the notification will be added again to it.